BinaryEncoding.md

```diff
@@ -147,28 +147,31 @@ A module may contain at most one import table section.
 | func_name | `uint32` | offset from the start of the module of the string representing the function name |
 
 
-### Functions section
+### Function Signatures section
 
-ID: `functions`
+ID: `function_signatures`
 
-The Functions section declares the functions in the module and must be preceded by a [Signatures](#signatures-section) section. A module may contain at most one functions section.
+The Functions Signatures section declares the signatures of all functions in the
```

Comment:
Sounds good; I'll put the note in the Function Bodies section, where it already talks about the length correspondence.